connectors/graphql/src/main/java/io/camunda/connector/graphql/GraphQLFunction.java

```diff
@@ -0,0 +1,157 @@
+import com.google.gson.Gson;
+import io.camunda.connector.api.annotation.OutboundConnector;
+import io.camunda.connector.api.error.ConnectorException;
+import io.camunda.connector.api.outbound.OutboundConnectorContext;
+import io.camunda.connector.api.outbound.OutboundConnectorFunction;
+import io.camunda.connector.common.auth.OAuthAuthentication;
+import io.camunda.connector.common.constants.Constants;
+import io.camunda.connector.common.model.CommonRequest;
+import io.camunda.connector.common.model.CommonResult;
+import io.camunda.connector.common.services.AuthenticationService;
+import io.camunda.connector.common.services.HTTPProxyService;
+import io.camunda.connector.common.services.HTTPService;
+import io.camunda.connector.graphql.components.GsonComponentSupplier;
+import io.camunda.connector.graphql.components.HttpTransportComponentSupplier;
+import io.camunda.connector.graphql.model.GraphQLRequest;
+import io.camunda.connector.graphql.model.GraphQLResult;
+import io.camunda.connector.graphql.utils.GraphQLRequestMapper;
+import io.camunda.connector.graphql.utils.JsonSerializeHelper;
+import io.camunda.connector.impl.config.ConnectorConfigurationUtil;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@OutboundConnector(
+    name = "GRAPHQL",
+    inputVariables = {"graphql", "authentication"},
+    type = "io.camunda:connector-graphql:1")
+public class GraphQLFunction implements OutboundConnectorFunction {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(GraphQLFunction.class);
+
+  private final Gson gson;
+  private final GsonFactory gsonFactory;
+  private final HttpRequestFactory requestFactory;
+
+  private final String proxyFunctionUrl;
+
+  public GraphQLFunction() {
+    this(ConnectorConfigurationUtil.getProperty(Constants.PROXY_FUNCTION_URL_ENV_NAME));
+  }
+
+  public GraphQLFunction(String proxyFunctionUrl) {
+    this(
+        GsonComponentSupplier.gsonInstance(),
+        HttpTransportComponentSupplier.httpRequestFactoryInstance(),
+        GsonComponentSupplier.gsonFactoryInstance(),
+        proxyFunctionUrl);
+  }
+
+  public GraphQLFunction(
+      final Gson gson,
+      final HttpRequestFactory requestFactory,
+      final GsonFactory gsonFactory,
+      final String proxyFunctionUrl) {
+    this.gson = gson;
+    this.requestFactory = requestFactory;
+    this.gsonFactory = gsonFactory;
+    this.proxyFunctionUrl = proxyFunctionUrl;
+  }
+
+  @Override
+  public Object execute(OutboundConnectorContext context)
+      throws IOException, InstantiationException, IllegalAccessException {
+    final var json = context.getVariables();
+    var connectorRequest = JsonSerializeHelper.serializeRequest(gson, json);
+    context.validate(connectorRequest);
+    context.replaceSecrets(connectorRequest);
+
+    return proxyFunctionUrl == null
```

Comment:
If we have Apache Commons lib here, consider `StringUtils.isBlank(proxyFunctionUrl)`

Reply:
Or even better - `String::isBlank` since we're using Java 11 here